Controller.php

```diff
@@ -259,6 +259,22 @@ public function callback()
 
         $user = $this->getUserByRemoteId("oidc", $providerUserId);
 
+
+	//auto linking
+	if ($settings->autoLinking->getValue() ) {
+	        $userModel = new Model();
+        	$matomoUser = $userModel->getUser($providerUserId);
+		if (! empty($matomoUser)) {
+			if (empty($user)) {
+				$this->linkAccount($providerUserId, $providerUserId);
+			}
+	
+        		$user = $this->getUserByRemoteId("oidc", $providerUserId);
+		}
+	}
+	//
```

Comment:
Formatting here is a little bit off.
Please use 4 space indentation as in the rest of the code.

Please add one or two inline comments describing what is happening